### PR TITLE
New localstack_host format

### PR DESCRIPTION
**Pull request by simonrw**
The configuration for the main LocalStack project is changing:

We are deprecating `HOSTNAME_EXTERNAL` and `LOCALSTACK_HOSTNAME` and in their place using `LOCALSTACK_HOST` with the format `<hostname>:<port>`. Since this library is used as part of LocalStack, we need to change the format of the `LOCALSTACK_HOST` variable here to match.

In addition we document the configuration as suggested in [issue #26](https://github.com/localstack/localstack-python-client/issues/26) and [PR #44](https://github.com/localstack/localstack-python-client/pull/44).

To complement https://github.com/localstack/localstack/pull/7893

### CODE REVIEW

File: localstack_client/config.py [hunk 1]
```diff
@@ -9,122 +8,122 @@
...
-    "waf": "{proto}://{host}:4640",
-    "wafv2": "{proto}://{host}:4640",
-    "config": "{proto}://{host}:4641",
-    "configservice": "{proto}://{host}:4641",
-    "mwaa": "{proto}://{host}:4642",
-    "fis": "{proto}://{host}:4643",
-    "meteringmarketplace": "{proto}://{host}:4644",
-    "transcribe": "{proto}://{host}:4566",
-    "mq": "{proto}://{host}:4566"
+_service_ports: Dict[str, int] = {
```
Comment (whummer) [inline localstack_client/config.py:13]: I'm inclined to remove this mapping altogether, as it has been painful and is no longer in use. This dates back to the [`0.11.x` era](https://github.com/localstack/localstack/issues/2983) (before we even introduced the edge port :) ), and is not working anymore anyways. v2 could be a good opportunity to get rid of `USE_LEGACY_PORTS`! 🧹

**Review by whummer — APPROVED** [review]
Thanks for tackling this @simonrw ! Can you please also add an entry to `CHANGELOG.md`, and bump the version in `setup.cfg`, so we can release this right away.. 👍 

As I mentioned, I think we could use the opportunity to get rid of the service port mapping altogether. This would make our lives easier, as we don't need to manually update the mapping when introducing new services! (which we currently still need to do..)

If we want to keep the changes minimal, we could also tackle this post v2, though (removing `USE_LEGACY_PORTS` should have no negative side-effects (as it is anyway no longer working with latest LocalStack), and could be done any time, even in a patch release).

File: localstack_client/config.py (hunk 1 above)
Comment (thrau) [reply localstack_client/config.py:11]: I'm not 100% sure what that will do to the controll paths of legacy s3 and lambda providers. We may still access them in some adapter code but not 100% sure

Comment (simonrw) [reply localstack_client/config.py:13]: Let's leave it in for now and revisit later, just in case :+1:

**Pull request merged**